# Profitable App Profiles for the App Store and Google Play Markets

The aim for this project is to find apps that are attractive to users and hence identify the type of apps that will generate the most revenue. As data analysts in a company that builds Android and iOs mobile apps, we help our developers to make data driven decisions with respect to the kind of apps they make.

### Let's get to know our data

In this project, we will be exploring two datasets, `Apple App Store` and `Google Play Store`. These datasets are only a sample since both markets have over 4 million apps which will require significant amount of time and money to collect. 

[Google Play Store](https://www.kaggle.com/lava18/google-play-store-apps) has approximately 10,000 Android apps.You can download the dataset [here](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv)

[The Apple App Store](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) has approximately 7,000 Android apps.You can download the dataset [here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)

We will first open and read our datasets into separate lists.

In [5]:
from csv import reader
#The Google Play data set 
open_file = open('googleplaystore.csv')
read_file = reader(open_file)
android_apps = list(read_file)
android_header = android_apps[0]
android_apps = android_apps[1:]

# The App Store data set 
open_file = open('AppleStore.csv')
read_file = reader(open_file)
ios_apps = list(read_file)
ios_header = ios_apps[0]
ios_apps = ios_apps[1:]

#android_apps[10472] = ['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']

We have separated the header from the apps data in each dataset.
- Header -> `android_header`, `ios_header`
- Actual dataset - > `android_apps`, `ios_apps` 

We assign the Google Play and Apple Store list of lists or dataset to `android_apps` and `ios_apps` respectively.

## Data Exploration 🚀

In [8]:
def explore_data(dataset, start, end, rows_and_columns=False):
    '''
    We will print a portion of the data, the number of rows and columns in our dataset
 
    Arg:
        dataset: list of lists
        start: integer representing start index of a slice from the data set
        end: integer representing the ending index of a slice
        rows_and_columns: Boolean, has False as default 
        
    Returns:
        None
    '''
    dataset_slice = dataset[start:end]    
    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

In [9]:
#print first few rows of each dataset and the number of columns and rows in each dataset
print('------------------------ANDROID APPS-----------------------------')
print(android_header)
print('\n')
explore_data(android_apps,0,2,True)

------------------------ANDROID APPS-----------------------------
['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


As you can see, the Google Play data set has 10,841 apps and 13 features that describe our data. Columns that maybe be useful to our analysis about the best apps in the Play Store: ` App`,  `Category`,  `Rating`,
`Reviews`,  `Installs`, `Type`,  `Price` and  `Genres`. 

If you need more information about the dataset's columns, you can check out [this page](https://www.kaggle.com/lava18/google-play-store-apps). 

In [11]:
print('------------------------IOS APPS---------------------------------')
print(ios_header) #column names for IOS
print('\n')
explore_data(ios_apps,0,3,True)

------------------------IOS APPS---------------------------------
['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns: 16


The IOS App dataset has 7,197 apps and 16 features that describe our data. These columns could help us in analysis of the best apps in the Apple App Store: `price`, `user rating`, `prime genre`, and `size bytes`. 

For more detailed information about the dataset's features if the column names are not descriptive enough check out [this page](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps). 

## Data cleaning

Before any analysis, we need to make sure that our data has the correct data that will not affect our analysis results.

This process is called data cleaning. It is often said that data scientists spend 80% of their time cleaning data, only 20% actually analyzing the cleaned data. This is not far from the truth in reality.


This data cleaning process, we will:

- Remove inaccurate data
- Remove duplicate app entries
- Remove non-English apps
- Isolating free apps

### Deleting incorrect data

From the Google Play data set [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion), we can see that from one of these [discussions](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015) that the row 10472, has a missing `Rating` column. 

Let's view this difference by printing this row (row 10472), the column's row (named as header) and another row from the dataset.

In [18]:
print(android_apps[10472])
print('\n')
print(android_header)
print('\n')
print(android_apps[0])

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


As you can see, the description of row #10472 for column "Category" is missing and it just skipped to the '1.9' rating description. This needs to be deleted.

In [19]:
# remove the row using del statement
#don't run this more than once, otherwise you'll delete more than one row
del android_apps[10472]

# Removing duplicate entries

When we explore the Google Play Store dataset further, we will notice that there are some 'doppelgangers' apps with duplicate entries.

For example, the Instagram app has 4 entries.

In [20]:
for app in android_apps:
    name  = app[0]
    if name == 'Instagram':
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


This is not the only duplicated app. As you can see from the code below, there are 1,181 cases where this occcurs.

## 1. Counting duplicates

In [21]:
duplicate_apps = []
unique_apps = []

for app in android_apps:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps: ',len(duplicate_apps))
print('\n')
print('Examples of duplicate apps: ',duplicate_apps[:15])

Number of duplicate apps:  1181


Examples of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


We however, can not count apps more than once when analyzing our data hence we need to remove the duplicate entries and keep only one entry per app.

If you examine the rows we printed for the Instagram app, the main difference happens on the fourth position of each row, which corresponds to the `number of reviews`.

The different numbers show the data was collected at different times. The row with the highest number of reviews shows the most recent and reliable data and therefore, will be kept.

## 2. Removing duplicates using a dictionary

We will remove the duplicates using the following steps:
- Creating a dictionary where each dictionary key is a unique app and the corresponding value is the highest number of reviews of that app
- Using the information stored in the dictionary, create a new data set which will have only one entry per app
- For each app, only the entry with the highest number of reviews will be selected

In [22]:
reviews_max = {}

for app in android_apps:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

When we inspect our dictionary, we expect to hold information of unique apps only which would amount to 9,658 entries. 
Which is the difference between our total dataset and the duplicated entries.

In [23]:
print('Expected length:', (len(android_apps)-len(duplicate_apps)))
print('Actual length:', len(reviews_max))

Expected length: 9659
Actual length: 9659


To remove the duplicate rows from the initial dataset using our dictionary:

- Create two lists, one to store the new cleaned data set and another to store only the app names e.g already_added

- Loop through the data set, obtaining the app name and reviews features for each app and store both in a variable 

- Inside the loop check if the reviews is equal to the number of maximum reviews of the app (number in the reviews_max dictionary for that app) and if the name is not already in the list of app names( i.e already_added), hence if both true:

    * Append the entire row to the cleaned data set which all the rows
    
    * Append the name of the app to the list containing app names   
 

In [24]:
android_clean = []
already_added = []

for app in android_apps:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)        

We expect that our new cleaned data set , `android_clean`, only has unique rows/apps which is expected to be 9,658 rows. We can inspect its length to confirm this.

We will also explore our new data set using our explore_data() function

In [25]:
print(len(android_clean))

9659


In [26]:
explore_data(android_clean, 0 , 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of columns: 13


# Removing non-English Apps

If you explore the data sets from both App stores, you will notice that there are apps with names suggesting they are directed towards a non-English speaking audience.

In [27]:
print(ios_apps[813][1])
print(ios_apps[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

爱奇艺PPS -《欢乐颂2》电视剧热播
【脱出ゲーム】絶対に最後までプレイしないで 〜謎解き＆ブロックパズル〜
中国語 AQリスニング
لعبة تقدر تربح DZ


In this case, we are only focusing on those with English names thus we will remove those with non-English names .

One way we can do this is by removing each app with a name containing a symbol that is not commonly used in English text. English text usually includes letters fom English alphabet, numbers composed of digits from 0 to 9, punctuation marks and other symbols such as (+, *, /).

These characters which are specific to English texts are encoded using the ASCII system where each character has corresponding number between 0 and 127. We can use the built in function `ord()` to compute its associated number.

From this system, we can build a function to detect whether a character belongs to the common English characters or not. If an app name contains a character that is greater than 127, then it's most likely the app has a non-English name.

In [28]:
def is_english(string):
    for character in string:
        if ord(character)>127:
            return False
    return True     

We can check if an app name is English or non-English using the function we created above    

In [29]:
is_english('Instagram')

True

In [30]:
is_english('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

In [31]:
is_english('Docs To Go™ Free Office Suite')
#ord('™')

False

In [32]:
is_english('Instachat 😜')
#ord('😜')

False

You will notice that the function does not correctly identify certain English apps like `'Docs To Go™ Free Office Suite'`,  `'Instachat 😜'` etc and may result in losing a lot of data due to incorrectly labeled English apps.

Therefore, we will change the function to check for more than three characters from the input string instead of one character.

In [33]:
def is_english(string):
    non_ascii = 0
    for character in string:
        if ord(character)>127:
            non_ascii+=1
    if non_ascii > 3:
        return False
    else:
        return True   

In [34]:
is_english('Docs To Go™ Free Office Suite')

True

In [35]:
is_english('Instachat 😜')

True

In [36]:
is_english('爱奇艺PPS -《欢乐颂2》电视剧热播')

False

We will use our function to filter out the Non-English apps from both datasets. 

In [37]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)

for app in ios_apps:
    name = app[1]
    if is_english(name):
        ios_english.append(app)            

In [38]:
explore_data(android_english, 0, 10, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M', '50,000+

You can notice now there are 9614 Android apps remaining.

In [39]:
explore_data(ios_english, 0, 10, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


['429047995', 'Pinterest', '74778624', 'USD', '0.0', '1061624', '1814', '4.5', '4.0', '6.26', '12+', 'Social Networking', '37', '5', '27', '1']


['282935706', 'Bible', '92774400', 'USD', '0.0', '985920', '5320', '4.5', '5.0', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']


['5538347

After removing the non-English app names, we are now left with 6,183 iOS apps. 

# Isolating free from non-free apps 

As mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue is the in-app ads.

So to continue on with our analysis, we will only isolate the free apps from both datasets. 

In [40]:
android_final_dataset = []
ios_final_dataset = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final_dataset.append(app)
        
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final_dataset.append(app)
        

In [41]:
print(len(android_final_dataset))
print(len(ios_final_dataset))

8864
3222


We're now left with 8864 android apps and 3222 ios apps. We will use this for our analysis

## Most common genres in the App Store and Google Play Markets

As mentioned before, our aim is to determine the kind of apps that are likely to attract more users because the revenue we collect is highly influenced by the number of people who use our apps.

Our validation strategy for an app idea is mostly comprised of 3 steps:
    1. Building a minimal Android version of the app and adding it to Google Play
    2. If the app has a good response from users, we add more features to it
    3. If it is profitable enough after 6 months, we could then build an iOS version of the app and add it to the App Store.
    
Our end goal is to add the decided app on both platforms but before then we need to find app profiles or genres that are successful on both markets.

To get a sense of the most common genres in each market, we will build a frequency table for the `Category` and `Genres` columns in the Google Play data sets and another using the `prime_genre` column in the iOS App data set.

#### Frequency Table

We will use two functions to analyze the frequency tables. One function will generate the frequency tables that show percentages while the other will display the percentages in a descending order.

In [46]:
def freq_table(dataset, index):
    table = {}
    total = 0
    for row in dataset:
        total+=1
        value = row[index]
        if value in table:
            table[value]+=1
        else:
            table[value]=1
    table_percentages = {}
    for key in table:
        percentage = (table[key]/total) * 100
        table_percentages[key] = percentage
    return table_percentages       

In [47]:
def display_table(dataset, index):
    table = freq_table(dataset, index) ##called the first function here
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

### Most common genres in the App Store

In [50]:
print('Frequency table of genres in ios apps:')
print('\n')
display_table(ios_final_dataset,-5) #for prime_genre column

Frequency table of genres in ios apps:


Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Yes, `Games` alone has taken more than half of the free apps followed up with the `Entertainment` Genre taking up only 7% of the free apps in the Apple App Store and in the third runners up, `Photo & Video` with close to 5%. 

We can also notice that more than 70% of the apps are designed for fun `(games, photo and video, social networking, sports, music)` as compared to those designed for practical purposes `(education, shopping, utilities, productivity, lifestyle)` which may be considered rarer.

Though there's a large number of fun apps, it does not imply that apps of this types have a large number of users.

It might be different as well for the educational or productivity apps, users expect them to be a more professional ad-free paid app with no ads that will bring distractions and hence might consider them to have the best user experience and features to help them study or work.

### Most common categories / genres in the Google Play Store

In [56]:
print('Frequency table of categories in android apps:')
print('\n')
display_table(android_final_dataset,1) #Category column

Frequency table of categories in android apps:


FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661

As you can see from the results, the top most common genre is Family and Games. If you think about it, Family might have a majority of kids games.  

Following these are productivity genre apps ( `Tools`,`Business`, `lifestyle`, `productivity`, `finance` etc)

This is different with those from the App Store market where  `practical apps` are more popular on Google Play store than Apple store.

In [59]:
print('Frequency table of genres in android apps:')
print('\n')
display_table(android_final_dataset, -4) #genre

Frequency table of genres in android apps:


Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.93637

Since the `Genre` column is more granular, we will focus on the `Category` column since it is more general.

We also noticed that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and fun apps.

### Most popular Apps